src/cache/cache.rs

```diff
@@ -203,9 +203,31 @@ pub trait Storage: Send + Sync {
 
 /// Get a suitable `Storage` implementation from the environment.
 pub fn storage_from_environment() -> Box<Storage> {
+    let endpoint_env_res = env::var("SCCACHE_ENDPOINT");
+    let mut endpoint = String::from("s3.amazonaws.com");
+    match endpoint_env_res {
+        Ok(ep) => endpoint = ep,
+        Err(e) => warn!("Failed to interpret endpoint: {:?}", e),
+    }
+    let region_env_res = env::var("SCCACHE_REGION");
```

Comment:
You need to that is how API is designed to handle this what is needed is a region cache to be implemented and used. This is how it is done in `minio` SDKs.  But since sccache is only meant to be used with one bucket at any given time it doesn't make sense to make this request instead users can happily provide this value.